### DISCUSSION TITLE
Binary installation on Alpine Linux

### DISCUSSION THREAD

**Discussion by bwillis** [projectdiscovery/katana#993] [Q&A]
I was trying to install katana on an Alpine Linux based image and I'm having some issues and was hoping someone could point me in the right direction. This is the Dockerfile I'm using:

```dockerfile
FROM alpine:3.20

# SET WORKDIR
WORKDIR /app

# Set the SHELL option -o pipefail
SHELL ["/bin/sh", "-o", "pipefail", "-c"]

# DEPENDENCIES
RUN apk -U upgrade --no-cache \
    && apk add --no-cache \
    unzip curl bash wget ca-certificates chromium \
    gcompat

# KATANA
RUN mkdir /tmp/katana-install \
    && curl -s https://api.github.com/repos/projectdiscovery/katana/releases/latest \
        | grep "browser_download_url.*linux_amd64.zip" \
        | cut -d : -f 2,3 \
        | tr -d \" \
        | xargs wget -q -O /tmp/katana-install/katana.zip \
    && unzip -d /tmp/katana-install/ /tmp/katana-install/katana.zip \
    && mv /tmp/katana-install/katana /usr/local/bin/katana \
    && chmod +x /usr/local/bin/katana \
    && rm -rf /tmp/katana-install

ENTRYPOINT [ "/app/entrypoint.sh" ]
```

It's worth noting that I've use this approach to install other projectdiscovery tools and didn't have any issues. It was a little hard discerning all of the installation requirements, but I have the chromium dependency and glibc through gcompat (if that's required). When I run this and attempt to invoke katana, I've gotten an error:

Docker:
```shell
rosetta error: failed to open elf at /lib64/ld-linux-x86-64.so.2
 Trace/breakpoint trap
```

Orbstack
```shell
OrbStack ERROR: Dynamic loader not found: /lib64/ld-linux-x86-64.so.2

This usually means that you're running an x86 program on an arm64 OS without multi-arch libraries.
To fix this, you can:
  1. Use an Intel (amd64) container to run this program; or
  2. Install multi-arch libraries in this container.

This can also be caused by running a glibc executable in a musl distro (e.g. Alpine), or vice versa.

For more details and instructions, see https://go.orbstack.dev/multiarch
```

## Workaround

It's worth noting the workaround for posterity and incase it gives any insight into the solution. By using multistage build from the published image, I can copy the binary into the Alpine Linux image and this works:

```dockerfile
FROM projectdiscovery/katana:latest as katana

FROM alpine:3.20
RUN apk -U upgrade --no-cache \
    && apk add --no-cache bind-tools ca-certificates chromium
COPY --from=katana /usr/local/bin/katana /usr/local/bin/katana

ENTRYPOINT ["/app/entrypoint.sh"]
```

Any help would be appreciated!

**Comment by GeorginaReeder** [comment]
Hi @bwillis , thanks for your question!

@dwisiswant0 , do we know what might be causing the installation issues here?

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
